File: backend-springboot/src/main/java/com/greenalgeria/zone/application/command/AddPhotoToZoneHandler.java
package com.greenalgeria.zone.application.command;

import com.greenalgeria.shared.cqrs.CommandHandler;
import com.greenalgeria.shared.exception.NotFoundException;
import com.greenalgeria.zone.application.*;
import com.greenalgeria.zone.domain.ZoneRepository;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@Component
@Transactional
public class AddPhotoToZoneHandler implements CommandHandler<AddPhotoToZoneCommand, Void> {

    private final ZoneRepository zoneRepository;

    public AddPhotoToZoneHandler(ZoneRepository zoneRepository) {
        this.zoneRepository = zoneRepository;
    }

    public Void handle(AddPhotoToZoneCommand command) {
        var zone = zoneRepository.findById(command.id()).orElseThrow(() -> new NotFoundException("Zone not found"));
        zone.addPhoto(command.photoUrl());
        zoneRepository.save(zone);
        return null;
    }

    @Override
    public Class<AddPhotoToZoneCommand> supportedCommand() {
        return AddPhotoToZoneCommand.class;
    }
}

File: backend-nestjs/src/modules/zones/application/commands/add-photo-to-zone.handler.ts
import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
import { Inject, NotFoundException } from '@nestjs/common';
import type { Cache } from 'cache-manager';
import { CACHE_MANAGER } from '@nestjs/cache-manager';
import { AddPhotoToZoneCommand } from './add-photo-to-zone.command';
import { ZoneRepository } from '../../infrastructure/zone.repository';

@CommandHandler(AddPhotoToZoneCommand)
export class AddPhotoToZoneHandler implements ICommandHandler<AddPhotoToZoneCommand> {
  constructor(
    private readonly repository: ZoneRepository,
    @Inject(CACHE_MANAGER) private readonly cache: Cache,
  ) {}

  async execute(command: AddPhotoToZoneCommand): Promise<void> {
    const zone = await this.repository.findById(command.zoneId);
    if (!zone) throw new NotFoundException(`Zone ${command.zoneId} not found`);

    zone.addPhoto(command.photoUrl);
    await this.repository.save(zone);
    await this.cache.del('zones:all');
    await this.cache.del(`zone:${command.zoneId}`);
  }
}
